### PR TITLE
refactor set_nest_at_index

### PR DESCRIPTION
**Pull request by Wilman2612**

### CODE REVIEW

**Comment by Wilman2612** [comment]
#1948

**Comment by Wilman2612** [comment]
about this error : 

   output for  set_nest_at_index  on run:  shapemismatch:valuetensorofshape[2]cannotbebroadcasttoindexingresultofshape[1]torchivy.set_nest_at_index(x,y,z)ivy.array([[1.,2.],[3.,4.]]){'a':0,'b':[1,2],'c':(4,5)}[['a','b','c'],['d','e','f'],['g',['H','i']]] 
output in docs : ivy.array([[1.,2.],[5.,6.]){'a':0,'b':[1,2],'c':(4,5)}[['a','b','c'],['d','e','f'],['g',['H','i']]]

because of this : shapemismatch:valuetensorofshape[2]cannotbebroadcasttoindexingresultofshape[1]torchivy. Looks like an error in the front end implementation of the array setter is not allowing to execute the set_nest_at_index function therefore the output turns out incorrect. Should i remove Array and NativeArray from the signature or is something that will be fixed?

File: ivy/functional/ivy/nest.py [hunk 1]
```diff
@@ -46,7 +46,10 @@ def prune_nest_at_index(nest, index):
         prune_nest_at_index(nest[index[0]], index[1:])
 
 
-def set_nest_at_index(nest, index, value):
+def set_nest_at_index(
+    nest: Union[ivy.Array, ivy.NativeArray, Dict, List], 
+    index: Tuple,
```
Comment (jiahanxie353) [inline ivy/functional/ivy/nest.py:51]: `index` could be more than `Tuple`

File: ivy/functional/ivy/nest.py [hunk 2]
```diff
@@ -46,7 +46,10 @@ def prune_nest_at_index(nest, index):
         prune_nest_at_index(nest[index[0]], index[1:])
 
 
-def set_nest_at_index(nest, index, value):
+def set_nest_at_index(
+    nest: Union[ivy.Array, ivy.NativeArray, Dict, List],
```
Comment (jiahanxie353) [inline ivy/functional/ivy/nest.py:50]: `nest` could also be a `ivy.Container`

**Review by jiahanxie353 — CHANGES_REQUESTED** [review]
Your current docstrings are right, and please refer to the [deep dive docstring examples](https://lets-unify.ai/ivy/deep_dive/12_docstring_examples.html) as this section mentions many rubrics. For example, you should give docstring example of `ivy.Container`

Also, there are issues raised from docstrings tests, and the error is raised from `backend/torch/general/scatter_nd`, which is  irrelevant to your changes. I'm trying to fix it.

**Comment by Wilman2612** [comment]
@jiahanxie353  I haven't included an container example because I think that is not a nestable function since is no implemented in ivy.container as specified in the function types and containers sections of the deep dive

**Comment by Wilman2612** [comment]
@jiahanxie353 I have included an example container. Is there anything else I need to amend?

**Comment by jiahanxie353** [comment]
> @jiahanxie353 I have included an example container. Is there anything else I need to amend?

@Wilman2612 it looks good to me. I will merge it.
Thanks very much for your contribution :)

**Pull request merged**